Diplom_3 (1.1)

firuzaminov opened the pull request.

firuzaminov committed:
Diplom_3 (1.1)

pomegranate282922 commented on src/main/java/pages/BurgerAssembling.java:
```diff
@@ -0,0 +1,49 @@
+        driver.findElement(FILLINGS_SECTION).isDisplayed();
+    }
+    @Step("Get current active text")
+    public String currentActiveText() {
```
Отлично, что анализ видимости делается через класс.

pomegranate282922 commented on src/test/java/Cabinet.java:
```diff
@@ -0,0 +1,44 @@
+    public void setUp() {
+        driver = new ChromeDriver();
+        driver.manage().window().maximize(); //вызов метода для открытия окна браузера во весь экран
+        driver.get("https://stellarburgers.nomoreparties.site/"); //открываем сайт
```
Все урлы должны быть вынесены как константы. Нужно исправить.

pomegranate282922 commented on src/main/java/pages/Registration.java:
```diff
@@ -0,0 +1,85 @@
+    private static final By LOGOUT_BUTTON = By.xpath("(//button[@type='button'])[1]");
+    private static final By USER_CABINET_BUTTON = By.xpath("//p[text()='Личный Кабинет']");
+    private static final By LOGIN_BUTTON = By.xpath("//button[text()='Войти в аккаунт']");
+    private static final By LOGIN_BUTTON_2 = By.xpath("//button[text()='Войти']");
```
Не используем магические цифры в именах. Нужно дать говорящее название исходя из бизнес логики. Нужно исправить.

pomegranate282922 commented on src/main/java/pages/Registration.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    @Step("Login button on main page click_2")
+    public void loginButtonOnMainPageClick_2() {
```
Тоже нужно убрать магические цифры в именах. Нужно исправить.

pomegranate282922 commented on src/test/java/Cabinet.java:
```diff
@@ -0,0 +1,44 @@
+    public void goToUserCabinetByUserCabinetButtonTest() {
+        Registration registration = new Registration(driver);
+        registration.loginButtonOnMainPageClick();
+        registration.inputEmail("qwe@qwe.com");
```
Так тесты очень зависят от состояния базы данных. Нужно создавать юзера перед тестом ( лучше через апи: так и быстрее, и надежнее) и обязательно удалять его после теста ( через апи). Данные юзера лучше рандомизировать.
Нужно исправить.

pomegranate282922 commented on src/test/java/OutFromAccountTest.java:
```diff
@@ -0,0 +1,33 @@
+        registration.userAccountButtonClick();
+        registration.logoutUser();
+        registration.loginButtonIsDisplayedCheck();
+    }
```
Аналогичное замечание про юзера нужно применить ко всем тестам. Нужно исправить.

pomegranate282922 commented on src/test/java/OutFromAccountTest.java:
```diff
@@ -0,0 +1,33 @@
+        registration.loginButtonOnMainPageClick();
+        registration.inputEmail("qwe@qwe.com");
+        registration.inputPassword("123456");
+        //loginPage.loginUser("arxipkin@gmail.com", "1234567");
```
Закомментированный код нужно удалить. Нужно исправить.

pomegranate282922 commented on src/test/java/RegistrationTest.java:
```diff
@@ -0,0 +1,52 @@
+
+
+
+
```
Обязательно удалить юзера. И лучше рандомизировать данные.
Нужно исправить.

pomegranate282922 reviewed (commented):
Работа хорошая! Отлично реализована POM, есть все тесты.
Но есть и замечания:
1. Нет помника и отчета allure.
2. Нет удаления созданных тестовых юзеров в тесте на регистрацию 
3. Тесты должны быть независимы от состояния базы данных- для этого создаём юзера перед тестом и удаляем после, а не используем хардкод 
4. Остальное по мелочи ( нейминг в основном)

firuzaminov committed:
add allure report

firuzaminov committed:
Diplom_3 (1.2) Создавал рандомного пользователя через API и удалял его

pomegranate282922 commented on src/main/java/pages/Registration.java:
```diff
@@ -0,0 +1,104 @@
+public class Registration {
+    public static final String REGISTER_PAGE_PATH = "https://stellarburgers.nomoreparties.site/register";
+    private final WebDriver driver;
+    private final By NAME = By.xpath(".//label[text()='Имя']/following-sibling::input");
```
Если поля нестатические - именуем их начиная со строчной буквы camelCase. И в любом случае в po или все поля-элементы статические, или все нестатические.
Нужно привести в соответствие.

pomegranate282922 commented on src/test/java/BurgerAssemblingTest.java:
```diff
@@ -0,0 +1,55 @@
+        BurgerAssembling burgerAssembling = new BurgerAssembling(driver);
+        burgerAssembling.saucesSectionClick();
+        burgerAssembling.bunsSectionClick();
+        assertEquals(burgerAssembling.currentActiveText(), "Булки");
```
Сначала идёт ожидаемое значение, потом актуальное. Нужно исправить везде.

pomegranate282922 commented on src/test/java/RegistrationTest.java:
```diff
@@ -0,0 +1,68 @@
+
+    @After
+    public void deleteUser() {
+        if (auth != null) {
```
Здесь auth всегда null. Нужно удалять юзера в обоих(!) кейсах.
Нужно исправить

firuzaminov replied:
В этом тесте я не создаю пользователя через API. Я просто генерю рандомные данные. Через API я удалю только при позитивной проверке когда пользователь реально создался. А как я через API удалю пользователя, который в конечном счете не создался?)
Поэтому реализовал вот тк

pomegranate282922 reviewed (commented):
Многое поправил- супер! Чуть осталось ещё замечаний.

firuzaminov committed:
Diplom_3 (1.4)
Исправил нейминг переменных, привел к одному виду
В BurgerAssembling привел в соответсвие ожидаемый и фактический результат
Реалзиовал удаление пользователя в двух кейсах в тесте про регистрацию
В некоторых тестах привел к одному виду ввод данных в инпуты
Все три замечания исправлены

firuzaminov committed:
Diplom_3 (1.5)
Выровнил методы в UserClient

pomegranate282922 commented on src/test/java/RegistrationTest.java:
```diff
@@ -0,0 +1,69 @@
+    @After
+    public void deleteUser() {
+        RestAssured.baseURI = URL_API;
+        userClient.deleteUserForRegistrationTest(email, password); //Создал еще один метод для удаления юзера в обоих кейсах
```
Да, именно это и хотелось. 
в принципе, чтобы не вызывать 2 раза метод, можно короткий пароль тоже записать в переменную password ( и делать присвоение в тестовом методе).

pomegranate282922 commented on src/main/java/client/UserClient.java:
```diff
@@ -0,0 +1,68 @@
+        } else return null;
+    }
+
+    public void deleteUserForRegistrationTest(String email, String password) {
```
Здесь лучше назвать метод без привязки к имени теста. Просто : deleteUserByCredentials

pomegranate282922 reviewed (commented):
Отлично, зачёт!
